src/jsdocUtils.js

```diff
@@ -186,20 +186,26 @@ const potentiallyEmptyNamepathTags = [
   'event',
   'callback',
   'class', 'constructor',
+  'extends', 'augments',
```

Comment:
Why are these being added here? I don't think these make much sense to be empty.

Reply:
I see that the test below ends up insisting they are not empty but note that the variable name here is `potentiallyEmptyNamepathTags`, so I think a new variable may need to be added to avoid confusion.

Reply:
The semantics here is that can omit namepath. `extends`/`augmens` can either contain a namepath or a type and must contain one of those.

I think the variables I used in the rule (`mustHaveType`, `mustHaveNamepath`, `mustHaveEither`) make more sense, but I was hesitant to change utility names.

If that is not a concer, I’d do something like `canHaveType`, `mustHaveType` and `mustHaveNamepathOrType` here. Or perhaps some configuration dictionary could be a good idea.

Reply:
I'd have to take a closer look, but I'm not against the PR changing the utility names in principle, as long as they are not being used (or are not expected to be used) with their old names/purpose.

Reply:
I went ahead and renamed utils along with their usage (most was here in validTypes).
Now their naming starts with `tagMustHave`/`tagMightHave` and ends with `Type`,`Namepath`,`EitherTagOrNamepath`.
I also moved `checkSeesForNamepaths` handling out of utils since this is a specifically `valid-types` option.